tasks/mpi/dudchenko_o_sleeping_barber/src/ops_mpi.cpp

```diff
@@ -0,0 +1,144 @@
+#include "mpi/dudchenko_o_sleeping_barber/include/ops_mpi.hpp"
+
+#include <deque>
+#include <functional>
+#include <string>
+#include <thread>
+
+using namespace std::chrono_literals;
+
+namespace dudchenko_o_sleeping_barber_mpi {
+
+bool TestMPISleepingBarber::pre_processing() {
+  internal_order_test();
+  result = -1;
+
+  if (world.rank() == 0) {
+    max_wait = taskData->inputs_count[0];
+  }
+
+  return true;
+}
+
+bool TestMPISleepingBarber::validation() {
+  internal_order_test();
+
+  if (world.rank() == 0) {
+    if (taskData->inputs_count.empty() || taskData->inputs_count[0] <= 0) {
+      return false;
+    }
+  }
+
+  if (world.rank() < 0 || world.rank() >= world.size()) {
+    throw std::runtime_error("Invalid rank.");
+  }
+
+  if (world.size() < 3) {
```

Comment:
Remove every world.size() check, algorithm doesnt work. Expression
{ if world.size() < 3 } always True in repository